autoPyTorch/utils/common.py

```diff
@@ -75,6 +76,25 @@ def __str__(self) -> str:
             self.hyperparameter, self.value_range, self.default_value, self.log)
 
 
+class autoPyTorchEnum(str, Enum):
+    """
+    Utility class for enums in autoPyTorch.
+    Allows users to use strings, while we internally use
+    this enum
+    """
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, autoPyTorchEnum):
+            return type(self) == type(other) and self.value == other.value
```

Comment:
Somehow this line is not passed through when you pass a list that contains the identical enum member.
For example, when you replace L87 with `raise NotImplementedError`, `APTEnum.x in [APTEnum.x]` should raise an error, but it does not.

Note that it raises an error when you do `APTEnum.x in [APTEnum.y]` or `APTEnum.x == APTEnum.x`.
Probably, `__contains__` method in `enum` or `str` or `list` is causing the issue, but I could not find out easily.